components/apps/src/lib.rs

```
@@ -373,23 +398,25 @@ impl<R: Runner> Apps<R> {
         #[cfg(feature = "ndef-app")]
         apps.push(&mut self.ndef).ok().unwrap();
 
-        // App 2: secrets
-        #[cfg(feature = "secrets-app")]
-        apps.push(&mut self.oath).ok().unwrap();
+        if self.migrated_successfully {
+            // App 2: secrets
+            #[cfg(feature = "secrets-app")]
+            apps.push(&mut self.oath).ok().unwrap();
```

I think it would be clearer to skip creating the relevant apps if migration failed instead of skipping them in `apdu_dispatch` and `ctaphid_dispatch`, similar to what we do with opcard on a config error.